JOSS submission draft (#361)
icesat2py/icepyx: doc/source/tracking/paper.md

---
title: 'icepyx'
tags:
  - Python
  - ICESat-2
  - LiDAR
  - elevation
  - community
  - cloud
authors:
  - name: Jessica Scheick^[Corresponding author] # note this makes a footnote
    orcid: 0000-0000-0000-0000
    affiliation: 1
  - name: Wei Ji Leong
    affiliation: 2
  - name: Kelsey Bisson
    affiliation: 3
  - name: Anthony Arendt
    affiliation: 4
  - name: Shashank Bhushan
    affiliation: 4
  - name: Zach Fair
    affiliation: 5
  - name: Raphael Hagen
    affiliation: 4
  - name: Sarah Hall
    affiliation: 1  
  - name: Scott Henderson
    affiliation: 4
  - name: Tom Johnson
    affiliation: 6
  - name: Friedrich Knuth
    affiliation: 7
  - name: Tian Li
    affiliation: 8
  - name: Zheng Liu
    affiliation: 4
  - name: Luis Lopez
    affiliation: 9
  - name: Romina Piunno
    affiliation: 10
  - name: Nitin Ravinder
    affiliation: 11
  - name: Landung "Don" Setiawan
    affiliation: 4
  - name: Trey Stafford
    affiliation: 9
  - name: Amy Steiker
    affiliation: 9

>   - name: Tyler Sutterley

Reviewer: ```suggestion
  - name: Tyler Sutterley
    orcid: 0000-0002-6964-1194
```

    affiliation: 4
  - name: JP Swinski
    affiliation: 12
  - name: Bruce Wallin
    affiliation: 9
  - name: Learn2Pheonix
    affiliation: 13

# Note: first three authors are driving the publication. Additional contributors/authors are listed in alphabetical order by last name. Anyone who also contributes to preparing (including reviewing) the JOSS submission will be moved into ABC order after the first three and before the non-publication contributors to icepyx.
  
affiliations:
 - name: University of New Hampshire, USA
   index: 1
 - name: Ohio State University, USA
   index: 2
 - name: University of Colorado Boulder, USA
   index: 3
 - name: , USA
   index: 4

>  - name: , USA

Reviewer: ```suggestion
 - name: University of Washington, USA
```

  - name: Raphael Hagen
    affiliation: 4
  - name: Sarah Hall
    affiliation: 1  
  - name: Scott Henderson
    affiliation: 4
  - name: Tom Johnson
    affiliation: 6
  - name: Friedrich Knuth
    affiliation: 7
  - name: Tian Li
    affiliation: 8
  - name: Zheng Liu
    affiliation: 4
  - name: Luis Lopez
    affiliation: 9
  - name: Romina Piunno
    affiliation: 10
  - name: Nitin Ravinder
    affiliation: 11
  - name: Landung "Don" Setiawan
    affiliation: 4
  - name: Trey Stafford
    affiliation: 9
  - name: Amy Steiker
    affiliation: 9
  - name: Tyler Sutterley
    orcid: 0000-0002-6964-1194
    affiliation: 4
  - name: JP Swinski
    affiliation: 12
  - name: Bruce Wallin
    affiliation: 9
  - name: Learn2Pheonix
    affiliation: 13

# Note: first three authors are driving the publication. Additional contributors/authors are listed in alphabetical order by last name. Anyone who also contributes substantially to preparing the JOSS submission will be moved into ABC order after the first three and before the non-publication contributors to icepyx. Non-responsive coauthors will be removed from the list since their permission to be included was not granted.

>   

Author: @weiji14 Does this answer your question? Happy to have a more in depth conversation about it (that goes for anyone!).

Reviewer 2: Ah yes, that's a good answer. There's also some notes on https://joss.readthedocs.io/en/latest/submitting.html#authorship which is worth a read.

affiliations:
 - name: University of New Hampshire, USA
   index: 1
 - name: The Ohio State University, USA
   index: 2
 - name: University of Colorado Boulder, USA
   index: 3
 - name: Institution, USA
   index: 4
 - name: University of Washington, USA
   index: 5
 - name: Institution, USA
   index: 6
 - name: Institution, USA
   index: 7
 - name: Institution, USA
   index: 8
 - name: Institution, USA
   index: 9
 - name: University of Toronto, Canada
   index: 10
 - name: Institution, USA
   index: 11
 - name: Institution, USA
   index: 12
 - name: Institution, USA
   index: 13
date: 1 September 2022
bibliography: icepyx_pubs.bib

# Optional fields if submitting to a AAS journal too, see this blog post:
# https://blog.joss.theoj.org/2018/12/a-new-collaboration-with-aas-publishing
aas-doi:
aas-journal:
---

# Summary

> icepyx is both a software library and a community composed of ICESat-2 data users, developers, maintainers, and the scientific community.

Reviewer 2: Do we need a citation for ICESat-2? E.g.:

- Markus, T., Neumann, T., Martino, A., Abdalati, W., Brunt, K., Csatho, B., Farrell, S., Fricker, H., Gardner, A., Harding, D., Jasinski, M., Kwok, R., Magruder, L., Lubin, D., Luthcke, S., Morison, J., Nelson, R., Neuenschwander, A., Palm, S., … Zwally, J. (2017). The Ice, Cloud, and land Elevation Satellite-2 (ICESat-2): Science requirements, concept, and implementation. Remote Sensing of Environment, 190, 260–273. https://doi.org/10.1016/j.rse.2016.12.029

Or maybe there's a more suitable one. Just occurred to me that there's no mention of what ICESat-2 actually does in this entire paper so far :joy:

Author: Good point. I added a short paragraph on IS2 and this ref. Thanks!

We are working together to develop a shared library of resources - including existing resources, new code, tutorials, and use-cases/examples - that simplify the process of querying, obtaining, analyzing, and manipulating ICESat-2 datasets to enable scientific discovery.

# Statement of need

> icepyx aims to provide a clearinghouse for code, functionality to improve interoperability, documentation, examples, and educational resources that tackle disciplinary research questions while minimizing the amount of repeated effort across groups utilizing similar datasets. icepyx also hopes to foster collaboration, open-science practices, and reproducible workflows by integrating and sharing resources.
> 
> icepyx began during the cryosphere-themed ICESat-2 Hackweek at the University of Washington in June 2019. At the event, a clear need for a collaborative, shared community space that combined and generalized the tooling (including code and examples past, present, and future) written by Hackweek participants, ICESat-2 Science Team members, and the data user community. This project combined and generalized the existing scripts into a unified framework, adding examples, documentation, and testing where necessary and making them accessible for everyone. The library and community continue to grow and evolve, adding new features and building scientific literacy in open-science, cloud computing, and collaborative development practices. icepyx is now a foundational tool for accessing and working with ICESat-2 data and is featured in multiple scientific publications[@Bisson:2021; @Fernando:2021; @Li:2020], presentations [@js2021agu; @js2020agu; @js2019agu], and Hackweeks [@2022_IS2-HW-tutorials; @2020_IS2-HW-tutorials].

Reviewer 2: Need to check if it works, but I'd suggest having each sentence on a separate line so that it's easier for people to make 'suggestion changes' for specific lines rather than the whole paragraph. Similar to what the Acknowledgement section is doing.

```suggestion
icepyx aims to provide a clearinghouse for code, functionality to improve interoperability, documentation, examples, and educational resources that tackle disciplinary research questions while minimizing the amount of repeated effort across groups utilizing similar datasets.
icepyx also hopes to foster collaboration, open-science practices, and reproducible workflows by integrating and sharing resources.

icepyx began during the cryosphere-themed ICESat-2 Hackweek at the University of Washington in June 2019.
At the event, a clear need for a collaborative, shared community space that combined and generalized the tooling (including code and examples past, present, and future) written by Hackweek participants, ICESat-2 Science Team members, and the data user community.
This project combined and generalized the existing scripts into a unified framework, adding examples, documentation, and testing where necessary and making them accessible for everyone.
The library and community continue to grow and evolve, adding new features and building scientific literacy in open-science, cloud computing, and collaborative development practices.
icepyx is now a foundational tool for accessing and working with ICESat-2 data and is featured in multiple scientific publications[@Bisson:2021; @Fernando:2021; @Li:2020], presentations [@js2021agu; @js2020agu; @js2019agu], and Hackweeks [@2022_IS2-HW-tutorials; @2020_IS2-HW-tutorials].
```

---
title: 'icepyx'
tags:
  - Python
  - ICESat-2
  - LiDAR
  - elevation
  - community
  - cloud
authors:
  - name: Jessica Scheick^[Corresponding author] # note this makes a footnote
    orcid: 0000-0002-3421-4459
    affiliation: 1
  - name: Wei Ji Leong
    orcid: 0000-0003-2354-1988
    affiliation: 2
  - name: Kelsey Bisson
    orcid: 0000-0003-4230-3467
    affiliation: 3
  - name: Anthony Arendt
    orcid: 0000-0003-0429-6905
    affiliation: 4
  - name: Shashank Bhushan
    affiliation: 4
  - name: Zach Fair
    affiliation: 5
  - name: Raphael Hagen
    affiliation: 4
  - name: Sarah Hall
    affiliation: 1  
  - name: Scott Henderson
    affiliation: 4
  - name: Tom Johnson
    affiliation: 6
  - name: Friedrich Knuth
    affiliation: 7
  - name: Tian Li
    affiliation: 8
  - name: Zheng Liu
    affiliation: 4
  - name: Luis Lopez
    affiliation: 9
  - name: Romina Piunno
    affiliation: 10
  - name: Nitin Ravinder
    affiliation: 11
  - name: Landung "Don" Setiawan

>     affiliation: 4

Reviewer 3: ```suggestion
    orcid: 0000-0002-1624-2667
    affiliation: 4
```

# Statement of need

icepyx aims to provide a clearinghouse for code, functionality to improve interoperability, documentation, examples, and educational resources that tackle disciplinary research questions while minimizing the amount of repeated effort across groups utilizing similar datasets.
icepyx also hopes to foster collaboration, open-science practices, and reproducible workflows by integrating and sharing resources.

The Ice, Cloud, and Land Elevation Satellite-2 (ICESat-2) [@is2] was launched by NASA in September 2018.
The laser altimeter on board the satellite emits green light and measures the time until each pulse is returned to the satellite's sensors.
This information is used to determine the surface height of the land, ice, snow, trees, water, clouds, etc. that the satellite is passing over.
The instruments provide close to 500 GB of data per day, allowing scientists to investigate the surface height of earth's features in unprecedented detail.

icepyx began during the cryosphere-themed ICESat-2 Hackweek at the University of Washington in June 2019.

> At the event, a clear need for a collaborative, shared community space that combined and generalized the tooling (including code and examples past, present, and future) written by Hackweek participants, ICESat-2 Science Team members, and the data user community.

Author: It's trying to convey the "active" nature of the space - we weren't just trying to put together resources that already existed, and we wanted it to be a space that would continue to grow and evolve beyond Hackweeks (and especially the first one in 2019)... any rewording ideas to better convey this?

> At the event, a clear need for a collaborative, shared community space that combined and generalized the tooling (including code and examples past, present, and future) written by Hackweek participants, ICESat-2 Science Team members, and the data user community.

Reviewer 2: Something like this? Feel free to rephrase.

```suggestion
At the event, there was a clear need for a collaborative, shared community space that combined and generalized the tooling, in the form of a one stop shop for material written by past, present, and future Hackweek participants, ICESat-2 Science Team members, and the data user community.
```

> At the event, a clear need for a collaborative, shared community space that combined and generalized the tooling (including code and examples past, present, and future) written by Hackweek participants, ICESat-2 Science Team members, and the data user community.

Reviewer 4: My 2c is I like the 'past, present, future' language as it's simple adn conveys the message well. I like what @weiji14 worked up here

> At the event, a clear need for a collaborative, shared community space that combined and generalized the tooling (including code and examples past, present, and future) written by Hackweek participants, ICESat-2 Science Team members, and the data user community.

Reviewer 4: could also say 'At the event, there was a clear need for a collaborative, shared community space that UNIFIED and generalized the tools and material written by past, present, and future Hackweek participants, ICESat-2 Science Team members, and the data user community.

> At the event, a clear need for a collaborative, shared community space that combined and generalized the tooling (including code and examples past, present, and future) written by Hackweek participants, ICESat-2 Science Team members, and the data user community.

Reviewer 4: I realized that too after I had submitted the suggestion :( I love how it reads now

    orcid: 0000-0003-1994-1153
    affiliation: 6
  - name: Scott Henderson
    orcid: 0000-0003-0624-4965
    affiliation: 4
  - name: Friedrich Knuth
    orcid: 0000-0003-1645-1984
    affiliation: 4
  - name: Tian Li
    orcid: 0000-0002-1577-4004
    affiliation: 7
  - name: Zheng Liu
    orcid: 0000-0003-4132-8136
    affiliation: 4
  - name: Romina Piunno
    affiliation: 8
  - name: Nitin Ravinder
    affiliation: 9
  - name: Landung "Don" Setiawan
    orcid: 0000-0002-1624-2667
    affiliation: 4
  - name: Tyler Sutterley
    orcid: 0000-0002-6964-1194
    affiliation: 4
  - name: JP Swinski
    affiliation: 5
  - name: Anubhav
    orcid: 0000-0003-4017-2862
    affiliation: 10

# Note: first three authors are driving the publication. Additional contributors/authors are listed in alphabetical order by last name. Anyone who also contributes substantially to preparing the JOSS submission will be moved into ABC order after the first three and before the "non-publication" contributors to icepyx. Non-responsive coauthors will be removed from the list since their permission to be included was not granted.

affiliations:
 - name: University of New Hampshire, USA
   index: 1

>  - name: The Ohio State University, USA

Reviewer 2: Changing my affiliation to DevSeed :seedling:

```suggestion
 - name: Development Seed, USA
```

A unified framework of code and documentated examples for downloading, reading, and visualizing ICESat-2 data that is well tested makes it more accessible for everyone to use.
The library and community continue to grow and evolve, adding new features and building scientific literacy in open-science, cloud computing, and collaborative development best practices.

icepyx is now a foundational tool for accessing and working with ICESat-2 data, responsible for nearly a quarter of all NASA data center granule downloads in 2022. The library is complimented by a series of other specialized tools for interacting with and obtaining ICESat-2 data. These include OpenAltimetry[@OA], a browser based web tool to visualize and download selected ICESat and ICESat-2 surface heights, SlideRule[@SR], a server-side framework to create cloud-based, on-demand customized data processing for ICESat-2 data, and multiple product-focused tools for scientific data analysis (e.g. PhoREAL, PhotonLabeler). These tools are described in more detail in the icepyx documentation's [ICESat-2 Resource Guide](https://icepyx.readthedocs.io/en/latest/community/resources.html).

> 

Author: @weiji14 Are you available to look over this updated text before I note the updates in the JOSS review?

Reviewer 2: One typo spotted. Just off the top of my head, I would maybe include Tyler's https://github.com/tsutterley/read-ICESat-2? I found a link to PhoREAL at https://github.com/icesat-2UT/PhoREAL but couldn't find one for PhotonLabeler, is it https://github.com/Oht0nger/PhoLabeler?

```suggestion
icepyx is now a foundational tool for accessing and working with ICESat-2 data, responsible for nearly a quarter of all NASA data center granule downloads in 2022. The library is complemented by a series of other specialized tools for interacting with and obtaining ICESat-2 data. These include OpenAltimetry[@OA], a browser based web tool to visualize and download selected ICESat and ICESat-2 surface heights, SlideRule[@SR], a server-side framework to create cloud-based, on-demand customized data processing for ICESat-2 data, and multiple product-focused tools for scientific data analysis (e.g. PhoREAL, PhotonLabeler). These tools are described in more detail in the icepyx documentation's [ICESat-2 Resource Guide](https://icepyx.readthedocs.io/en/latest/community/resources.html).

```
On the last sentence, I feel like there was a good NASA Earthdata or NSIDC page that lists out all the ICESat-2 data tools, but can't seem to find it anymore.

Reviewer 4: We may not need to include this but the Bathy working group has a GitHub site with tools too - https://github.com/ICESat2-Bathymetry/Information

Jessica, I found a separate tools link from the one you sent - https://nsidc.org/data/user-resources/help-center/icesat-2-tools-and-services

Author: > We may not need to include this but the Bathy working group has a GitHub site with tools too - https://github.com/ICESat2-Bathymetry/Information
> 
> Jessica, I found a separate tools link from the one you sent - https://nsidc.org/data/user-resources/help-center/icesat-2-tools-and-services

Thanks - these are great resources, Kelsey! I am adding them to #412.

Reviewer 2: Yes, https://nsidc.org/data/icesat-2/tools was the one I was looking for. Thanks for updating the draft pdf and opening https://github.com/icesat2py/icepyx/issues/412!
